src/ffi/ios.rs

```diff
@@ -4,19 +4,90 @@ use core_foundation::{
 };
 use http::Request;
 use libc::c_char;
+use tracing::info;
+use url::Url;
 
 use crate::{client::Client, DirectoryCache};
+use core_foundation::data::{CFDataRef, CFData};
 
 fn setup_logger() {
     let subscriber = tracing_fmt::FmtSubscriber::builder()
-        .with_max_level(tracing::Level::DEBUG)
+        .with_max_level(tracing::Level::TRACE)
         .with_writer(std::io::stderr)
         .finish();
     tracing::subscriber::set_global_default(subscriber).expect("to be the only logger");
 }
 
+#[no_mangle]
+pub unsafe extern "C" fn call_arti(
+    method_c: *const c_char,
+    url_c: *const c_char,
+    _headers_c: *const c_char,
+    _body_c: *const c_char,
```

Comment:
```suggestion
    _body_c: *const u8,
```
I think this should be byte not char

Reply:
It's all JSON now...